File: src/lc/L-121_Best_Time_to_Buy_and_Sell_Stock/python/v1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        days = len(prices)
        max_profit = 0

        for i in range(days - 1):
            for j in range(i + 1, days - 1):
                max_profit = max(max_profit, prices[j] - prices[i])

        return max_profit

File: src/lc/L-121_Best_Time_to_Buy_and_Sell_Stock/java/v1.java
class Solution {
        public int maxProfit(int[] prices) {
                int max_profit = 0;
                int lowest_price = Integer.MAX_VALUE;

                for (int day = 0; day < prices.length; day++) {
                        int current_price = prices[day];

                        lowest_price = Math.min(lowest_price, current_price);

                        int current_max_profit = current_price - lowest_price;
                        max_profit = Math.max(max_profit, current_max_profit);
                }

                return max_profit;
        }
}